org.envirocar.app/src/org/envirocar/app/views/carselection/CarSelectionActivity.java

```diff
@@ -262,13 +264,21 @@ public void onDeleteCar(Car car) {
                                             showBackgroundImage();
                                         }
                                     }
+                                    if (mSelectedButton != null) {
+                                        mSelectedButton.setChecked(false);
+                                        // making the mSelectedButton null so there are no references of old deleted car
+                                        makeSelectedButtonNull(mSelectedButton);
+                                    }
                                     // then remove it from the list and show a snackbar.
                                     mCarListAdapter.removeCarItem(car);// Nothing to do on cancel
-
                                 })
                                 .setNegativeButton(R.string.cancel,null)
                                 .show();
                     }
+                    public void makeSelectedButtonNull(RadioButton mSelectedButton)
+                    {
+                        mSelectedButton = null;
+                    }
```

Comment:
Please, don't do something like this. Defining a method that contains a one-liner is such a bad pattern.

Reply:
sorry, but I had to do that because the inner class expression only takes final variables, maybe any suggestion to avoid this here? I will surely take care of this next time

Reply:
Ok, I see the problem. However, I think there is no need to set `mSelectedButton` to `null`. Setting the checked state to `false` should be sufficient. Give it a try.

Reply:
thank you, I have tested the code, if you get any problem I would sure test again :)